internal/app/mapper/mapper.go

```diff
@@ -124,20 +124,25 @@ func isNodeMatch(matchPredicate *matchPredicate, req transport.Request) (bool, e
 	if predicate == nil {
 		return false, nil
 	}
-	switch predicate.GetField() {
-	case aggregationv1.NodeFieldType_NODE_CLUSTER:
-		return compare(predicate, req.GetCluster())
-	case aggregationv1.NodeFieldType_NODE_ID:
-		return compare(predicate, req.GetNodeID())
-	case aggregationv1.NodeFieldType_NODE_LOCALITY_REGION:
-		return compare(predicate, req.GetRegion())
-	case aggregationv1.NodeFieldType_NODE_LOCALITY_ZONE:
-		return compare(predicate, req.GetZone())
-	case aggregationv1.NodeFieldType_NODE_LOCALITY_SUBZONE:
-		return compare(predicate, req.GetSubZone())
-	default:
-		return false, fmt.Errorf("RequestNodeMatch does not have a valid NodeFieldType")
+
+	// By construction only one of these is set at any point in time, so checking one by one
```

Comment:
@jyotimahapatra , do you know if there's a way to preserve the switch-like functionality here? From a correctness perspective my comment is true, albeit less performant than the old code.

Reply:
relying on short-circuiting is not better in terms of perf than what I'm proposing here.

Reply:
i think what we have here works.